engine/src/flutter/display_list/utils/dl_matrix_clip_tracker.cc

```diff
@@ -97,6 +97,37 @@ void DisplayListMatrixClipState::clipRRect(const DlRoundRect& rrect,
   }
 }
 
+void DisplayListMatrixClipState::clipRSuperellipse(
+    const DlRoundSuperellipse& rse,
+    DlClipOp op,
+    bool is_aa) {
+  DlRect bounds = rse.GetBounds();
+  if (rse.IsRect()) {
+    return clipRect(bounds, op, is_aa);
+  }
```

Comment:
Also IsOval -> clipOval

Reply:
I just realized that same optimization is missing from clipRRect and I'm not sure why...

Reply:
I've changed it to Oval.

> I just realized that same optimization is missing from clipRRect and I'm not sure why...

I could add it in this PR but I think it would be better in a separate PR. :)

Reply:
In that case do you think I should remove the oval check just to be super safe until it's found out? Once we know the reason we're going to need a PR to add it anyway.

Reply:
It couldn't hurt and then add a bug to add these in so we don't forget about it.

One thing about detecting an RSE as an oval - since the curvature of these doesn't follow a conic shape - can an RSE ever actually be an oval? An rrect can because its curvature is a pure conic, but an RSE is blended and so even if the radii span the space, it will never be a true oval AFAIK...?

Reply:
Yes, it will. As the parameter radius approaches 1/2 width or height, the superellipse segements become shorter and eventually 0, making the entire shape a circle or oval.

Reply:
* When height == width and radius == 1/2 height, then RSE becomes a circle.
* When height != width and xRadius == 1/2 width and yRadius == 1/2 height, the RSE becomes an ellipse.
* When only xRadius == 1/2 width and not for the height, the top and bottom parts are circular but not the other rest.

Reply:
I've removed the oval shortcut in RSE and created an issue for this: https://github.com/flutter/flutter/issues/163717